### ISSUE TITLE
some installation problems and file or directory not found

### ISSUE THREAD

**Issue by joyhuang9473** [progamesigner/dotfiles#1]
When I execute the install.sh, there are some message below:

```
/Users/user1/dotfiles/ruby/install.sh: line 3: ((: $+commands[brew] : syntax error: operand expected (error token is "$+commands[brew] ")
 [WAIT] ★ What is your name for git:
 [WAIT] ★ What is your email for git:
/Users/user1/dotfiles/git/install.sh: line 12: ((: $+commands[brew] : syntax error: operand expected (error token is "$+commands[brew] ")
 [INFO] ➜ Linking /Users/user1/.gitignore ...
```

```
/Users/user1/dotfiles/zsh/plugins/z/z.plugin.zsh:source:5: no such file or directory: /Users/user1/dotfiles/zsh/plugins/z/z/z.sh
/Users/user1/.zshrc:source:40: no such file or directory: /Users/user1/dotfiles/zsh/plugins/k/k.plugin.zsh
/Users/user1/.zshrc:source:40: no such file or directory: /Users/user1/dotfiles/zsh/plugins/zsh-syntax-highlighting/zsh-syntax-highlighting.plugin.zsh
/Users/user1/.zshrc:source:40: no such file or directory: /Users/user1/dotfiles/zsh/plugins/zsh-history-substring-search/zsh-history-substring-search.plugin.zsh
/Users/user1/.zshrc:source:40: no such file or directory: /Users/user1/dotfiles/zsh/plugins/bd/bd.plugin.zsh
/Users/user1//dotfiles/zsh/plugins/gpg-agent/gpg-agent.plugin.zsh:21: command not found: start_agent_nossh
/Users/user1/.zshrc:source:40: no such file or directory: /Users/user1/dotfiles/zsh/plugins/gitignore/gitignore.plugin.zsh
/Users/user1/.zshrc:source:40: no such file or directory: /Users/user1/dotfiles/zsh/plugins/caniuse/caniuse.plugin.zsh
/Users/user1/dotfiles/zsh/plugins/battery/battery.plugin.zsh:58: parse error near `]]'
align-center-indent:3: read-only variable: _
align-left-padding:3: read-only variable: _
```

Maybe this is just some minor errors.

**Comment by progamesigner** [comment]
These actually two issues. The first one is because I use ZSH snytax for Bash and I would fix them. The second one that you should use `git submodule update --init --recursive` to get some plugins.

**Comment by progamesigner** [comment]
The issue was fixed in 0bc7a03.
